### ISSUE TITLE
Built-in SSH server fails install

### ISSUE THREAD

**Issue by jodlajodla** [gogits/gogs#3495]
- Gogs version (or commit ref): v0.9.71 @ 2016-08-10 (amd64 binary)
- Git version: 2.7.4
- Operating system: Ubuntu 16.04.1 LTS
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes
  - [ ] No
  - [x] Not relevant
- Log gist:
```
panic: listen tcp 0.0.0.0:2222: bind: address already in use

goroutine 159 [running]:
panic(0x10987c0, 0xc8210425f0)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/gogits/gogs/modules/ssh.listen(0xc820475a20, 0x8ae)
        /home/vagrant/gopath/src/github.com/gogits/gogs/modules/ssh/ssh.go:116 +0x152
created by github.com/gogits/gogs/modules/ssh.Listen
        /home/vagrant/gopath/src/github.com/gogits/gogs/modules/ssh/ssh.go:183 +0x711
```

## Description

I wrote script for auto install of Gogs along with nginx. The script place prepared config into .../custom/conf and already contains DB info, repo root, log dir, ... When I open install URL, fill in additional data and click Install (it doesn't matter if request goes through nginx or directly to Gogs) there is some little load and after that I get error 502 Bad gateway (nginx) or connection failed (Gogs directly). This happens only in case that config contains options for built-in server:
```
START_SSH_SERVER = true
SSH_LISTEN_PORT  = 2222
```

If I remove those two lines from config then everything works well as expected. I figured out that built-in SSH server is started everytime when Gogs is started - also when install wasn't completed. So in case that there is no change to SSH port install fails because SSH server is already running from before on this port. The solution would probably be to disable SSH server on install or shut it down on install request and start it again after everything from install is in place.

**Comment by 0x9999** [comment]
Based on the error you posted, there is something alredy listening on port 2222 and gogs can't listen on it. Try to use a different port or disable/remove the service which is already listening on that port so gogs can listen on it.

**Comment by jodlajodla** [comment]
Yes, on this port is built-in server already listening because it is started when Gogs initially starts (and shows install page). So it seems that Gogs is somehow restarted during install (when Install button is clicked), but it doesn't stop built-in SSH server and wants to start it again on the same port, so install fails. There is 100% nothing else listening on that port at the time, because I tried with few different ports in different ranges and everytime I got this error. Also if I don't specify those two options mentioned above when Gogs initially starts for install, everything works as expected - I can add those two lines once install is finished and then Gogs works without problems.
